Jenkinsfile

```diff
@@ -0,0 +1,319 @@
+pipeline {
+    agent {
+        node {
+            label 'maven'
+        }
+    }
+    parameters{ 
+        string (defaultValue: 'notinuse', name:'OPENSHIFT_HOST', description:'open shift cluster url')
+        string (defaultValue: 'notinuse', name:'OPENSHIFT_TOKEN', description:'open shift token')
+        string (defaultValue: 'docker-registry.default.svc:5000', name:'IMAGE_REGISTRY', description:'open shift token')
+        string (defaultValue: 'rh-dev', name:'IMAGENAMESPACE', description:'name space where image deployed')
+        string (defaultValue: 'all', name:'DEPLOY_MODULE', description:'target module to work on')
+        string (defaultValue: 'rh-dev', name:'CICD_PROJECT', description:'build or development project')
+        string (defaultValue: 'rh-test', name:'TEST_PROJECT', description:'Test project')
+        string (defaultValue: 'rh-prod', name:'PROD_PROJECT', description:'Production project')
+        string (defaultValue: 'dbuser', name:'MYSQL_USER', description:'My Sql user name')
+        string (defaultValue: 'password', name:'MYSQL_PWD', description:'My Sql user password')
+    }
+    stages {
+        stage('Wait for user to select module to build.') {
+            steps {
+                script {
+                    try {
+                        timeout (time:600, unit:'SECONDS') {
+                            env.userSelModule = input(id: 'userInput', message: 'Please select which module to bulid?',
+                            parameters: [[$class: 'ChoiceParameterDefinition', defaultValue: 'strDef', 
+                               description:'describing choices', name:'nameChoice', choices: "Gateway\nFisUser\nFisAlert\nUI\nAll"]
+                            ])
+                        }
+                    } catch (exception) {
+                      env.userSelModule='All'    
+                    }
+        
+                    println("User selected module " + env.userSelModule);
+                }
+            }
+        }
+        stage('Build maingateway-service') {
+            environment {
+                serviceName = 'maingateway-service'
+            }
+            when {
+                expression {
+                    env.userInput == 'Gateway' || env.userInput == 'All'
+                }
+            }
+            steps {
+                echo "Building.. ${serviceName} "
+                build(env.serviceName)
+
+                echo "Deploying ${serviceName} to ${CICD_PROJECT}"
+                deploy(env.serviceName, params.CICD_PROJECT, params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, params.MYSQL_USER, params.MYSQL_PWD)
```

Comment:
Does these all parameters required to deploy gateway service.?

Reply:
no, changed.